File: CodingChallengeSolutionsC++/2176. Count Equal and Divisible Pairs in an Array/2176.CountEqualandDivisiblePairsinanArray.py
class CountEqualandDivisiblePairsinanArray:
    def countPairs(self, nums: List[int], k: int) -> int:
        if len(nums) == 1:
            return 0

        counter = 0
        for x in range(0, len(nums)):
            for y in range(x+1, len(nums)):
                if nums[x] == nums[y]:
                    if (x*y) % k == 0:
                        counter = counter + 1

        return counter

File: CodingChallengeSolutionsC++/2176. Count Equal and Divisible Pairs in an Array/2176.CountEqualandDivisiblePairsinanArray.js
var countPairs = function (nums, k) {
    counter = 0;

    for (x = 0; x < nums.length; ++x) {
        for (y = x + 1; y < nums.length; ++y) {
            if (nums[x] == nums[y]) {
                if ((x * y) % k == 0)
                    ++counter;
            }
        }
    }

    return counter;
};